File: 2022/rust/src/main.rs
mod days;
use std::{env, fmt::Display, fs};

// https://adventofcode.com/2022/day/<DAY>

// For first challenge: cargo run -- 1
// For second challenge: cargo run -- 2

fn main() {
    let args: Vec<String> = env::args().collect();

    if args.len() != 3 {
        println!("Usage: cargo run -- <day number> <puzzle part>");
        return;
    }

    let day: i32 = args[1].parse().expect("Expected a day number");
    let part: i32 = args[2].parse().expect("Expected the puzzle part");
    let nice_day = format!("{:0>2}", day);
    let file_name = format!("data/day{}.txt", nice_day);

    let contents = fs::read_to_string(&file_name)
        .expect(format!("Something went wrong reading the file: {}", &file_name).as_str());

    let answer = match day {
        1 => days::day01::process(part, contents).to_string(),
        2 => days::day02::process(part, contents).to_string(),
        3 => days::day03::process(part, contents).to_string(),
        4 => days::day04::process(part, contents).to_string(),
        5 => days::day05::process(part, contents),
        6 => days::day06::process(part, contents).to_string(),
        _ => {
            println!("Invalid day: {}", day);
            "ERR".into()
        }
    };

    println!(
        "The answer for day {} part {} is: {}",
        nice_day, part, answer
    );
}

File: 2022/python/src/main.py
import sys
import os
from importlib import import_module


def main():
    if len(sys.argv) != 3:
        print("Usage: python main.py <day> <part>")
        sys.exit(1)

    day = int(sys.argv[1])
    part = int(sys.argv[2])
    nice_day = str(day).zfill(2)
    file_name = os.path.join("data", f"day{nice_day}.txt")

    data = ""
    with open(file_name, "r") as f:
        data = f.read()

    # import the module for the day
    module = import_module(f"days.day{nice_day}")
    result = module.main(part, data)

    print(f"The answer for day {nice_day} is: {result}")


if __name__ == "__main__":
    main()
